### ISSUE TITLE
Conversion Error of a DATE column when reading a file to insert in table

### ISSUE THREAD

**Issue by fjossandon** [duckdb/duckdb#13660]
### What happens?

When reading from a tab-separated file using read_csv to insert into a previously created empty table, the execution gives a Conversion Error in a DATE value, although it's valid. The following is the original exception:
```
Conversion Error: CSV Error on Line: 1
Original Line: 6	Esta es una prueba hecha por Felipe	Esta es una prueba hecha por Felipe	1	s3://internal-files/6/2020-11-19_14.29		2020-10-28								0.1.4	False		False
Error when converting column "created_at". Could not convert string "2020-10-28" to 'DATE'
Column created_at is being converted as type DATE
This type was auto-detected from the CSV file.
Possible solutions:
* Override the type for this column manually by setting the type explicitly, e.g. types={'created_at': 'VARCHAR'}
* Set the sample size to a larger value to enable the auto-detection to scan more values, e.g. sample_size=-1
* Use a COPY statement to automatically derive types from an existing table.
  file=input.tsv
  delimiter = 	 (Set By User)
  quote = " (Auto-Detected)
  escape = " (Auto-Detected)
  new_line = \r\n (Auto-Detected)
  header = false (Set By User)
  skip_rows = 0 (Auto-Detected)
  date_format = %y.%m.%d (Auto-Detected)
  timestamp_format =  (Auto-Detected)
  null_padding=0
  sample_size=20480
  ignore_errors=false
  all_varchar=0
The Column types set by the user do not match the ones found by the sniffer. 
Column at position: 5 Set type: DATE Sniffed type: VARCHAR
Column at position: 16 Set type: INTEGER Sniffed type: VARCHAR
```

### To Reproduce

The following is the content for a test file to reproduce (`test.tsv`):
```
6	Esta es una prueba hecha por Felipe	Esta es una prueba hecha por Felipe	1	s3://internal-files/6/2020-11-19_14.29		2020-10-28								0.1.4	False		False
```

The following is the script to reproduce. Create the test file and update the paths at the beginning:

```python
from pathlib import Path

import duckdb

duckdb_path = Path("duckdb.db")
input_path = Path("test.tsv")

create_table = """
CREATE TABLE "_test_table"
    (id int4, description varchar, comment varchar, x_version int8, s3_path varchar, eta date, created_at date, a_id int4, b_id int4, c_id int4, d_id int4, e_id int4, version int4, is_std bool, y_version varchar, is_a bool, f_id int4, is_b bool)
;
"""
insert_query = f"""
        INSERT INTO "_test_table"
        SELECT * FROM read_csv(
            '{input_path}',
            delim = '	',
            header = false,
            columns = {{'id': 'int4', 'description': 'varchar', 'comment': 'varchar', 'x_version': 'int8', 's3_path': 'varchar', 'eta': 'date', 'created_at': 'date', 'a_id': 'int4', 'b_id': 'int4', 'c_id': 'int4', 'd_id': 'int4', 'e_id': 'int4', 'version': 'int4', 'is_std': 'bool', 'y_version': 'varchar', 'is_a': 'bool', 'f_id': 'int4', 'is_b': 'bool'}}
        );
"""

duckdb_path.unlink(missing_ok=True)
conn_duck = duckdb.connect(str(duckdb_path))

print("#####")
print(
    conn_duck.query(
        f"SELECT Prompt FROM sniff_csv('{input_path}',header = false);"
    ).to_df()["Prompt"].to_list()
)

print("#####")
print("#####")
conn_duck.execute(create_table)
try:
    conn_duck.execute(insert_query)
except Exception as e:
    print(e)

print("#####")
print("#####")
print(conn_duck.query("SELECT * FROM '_test_table';"))
```
Since I only put 1 line (the original has many more), and that line example has lots of NULL values, there will be more "type do not match" warnings in the output. But the error remains:
```
Conversion Error: CSV Error on Line: 1
Original Line: 6	Esta es una prueba hecha por Felipe	Esta es una prueba hecha por Felipe	1	s3://internal-files/6/2020-11-19_14.29		2020-10-28	0.1.4	False		False
Error when converting column "created_at". Could not convert string "2020-10-28" to 'DATE'

Column created_at is being converted as type DATE
```
The "2020-10-28" is a valid DATE, matching the type I configured in the read_csv arguments, but it still crashes.

### OS:

Ubuntu 23.10

### DuckDB Version:

1.0.0

### DuckDB Client:

Python 3.11.6

### Full Name:

Francisco Ossandon

### Affiliation:

Biome Makers

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by fjossandon** [comment]
I found something else, if I add `auto_detect=false` to the read_csv arguments, the error goes away and the insert is made correctly. So it looks like the default auto-detect is causing the conversion error somehow, maybe it conflicts with the type I specified with the `columns` argument??

**Comment by hawkfish** [comment]
I think the problem is that the auto-detect found dot-separated dates:

```
  date_format = %y.%m.%d (Auto-Detected)
```

We don't currently support multiple date formats for CSV reading, but you can use the [`try_strptime`](https://duckdb.org/docs/sql/functions/timestamp.html#try_strptimetext-format-list) function with a list of formats to convert the strings after you read them in.

**Comment by szarnyasg** [comment]
@fjossandon can you please try running this on latest [nightly build](https://duckdb.org/docs/installation/?version=main&environment=python) of DuckDB?

```bash
pip install duckdb --pre --upgrade
python your_script.py
```

This results in:

```text
#####
['FROM read_csv(\'test.tsv\', auto_detect=false, delim=\'\t\', escape=\'"\', new_line=\'\\n\', skip=0, comment=\'\x00\', columns={\'column00\': \'BIGINT\', \'column01\': \'VARCHAR\', \'column02\': \'VARCHAR\', \'column03\': \'BIGINT\', \'column04\': \'VARCHAR\', \'column05\': \'VARCHAR\', \'column06\': \'DATE\', \'column07\': \'VARCHAR\', \'column08\': \'VARCHAR\', \'column09\': \'VARCHAR\', \'column10\': \'VARCHAR\', \'column11\': \'VARCHAR\', \'column12\': \'VARCHAR\', \'column13\': \'VARCHAR\', \'column14\': \'VARCHAR\', \'column15\': \'BOOLEAN\', \'column16\': \'VARCHAR\', \'column17\': \'BOOLEAN\'}, dateformat=\'%Y-%m-%d\', header=false);']
#####
#####
#####
#####
┌───────┬──────────────────────┬──────────────────────┬───────────┬──────────────────────┬──────┬────────────┬───┬───────┬─────────┬─────────┬───────────┬─────────┬───────┬─────────┐
│  id   │     description      │       comment        │ x_version │       s3_path        │ eta  │ created_at │ … │ e_id  │ version │ is_std  │ y_version │  is_a   │ f_id  │  is_b   │
│ int32 │       varchar        │       varchar        │   int64   │       varchar        │ date │    date    │   │ int32 │  int32  │ boolean │  varchar  │ boolean │ int32 │ boolean │
├───────┼──────────────────────┼──────────────────────┼───────────┼──────────────────────┼──────┼────────────┼───┼───────┼─────────┼─────────┼───────────┼─────────┼───────┼─────────┤
│     6 │ Esta es una prueba…  │ Esta es una prueba…  │         1 │ s3://internal-file…  │ NULL │ 2020-10-28 │ … │  NULL │    NULL │ NULL    │ 0.1.4     │ false   │  NULL │ false   │
├───────┴──────────────────────┴──────────────────────┴───────────┴──────────────────────┴──────┴────────────┴───┴───────┴─────────┴─────────┴───────────┴─────────┴───────┴─────────┤
│ 1 rows                                                                                                                                                       18 columns (14 shown) │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

**Comment by fjossandon** [comment]
Hello again!
> I think the problem is that the auto-detect found dot-separated dates:

@hawkfish Ah, I see... you mean `y_version` column value "0.1.4". That is actually a code version "major.minor.patch", so I put varchar when setting it in the Create table and the "columns" argument of the read_csv. I have had several issues with types regarding different columns with arbitrary text or too many NULLs and so on (type-related errors on DuckDB or Pandas), that's why I got the idea of getting the data types directly from the database `information_schema.columns` and passing that explicitly to DuckDB when manipulating data to make it more robust. When writing this issue, I thought that the "auto_detect" would only fill in the values that I did not specify myself (columns types in this case), so it would only guess the format from the columns I set as Date, but this was not being considered.

----------

> @fjossandon can you please try running this on latest [nightly build](https://duckdb.org/docs/installation/?version=main&environment=python) of DuckDB?

@szarnyasg Yes, I just installed the nightly build with your instructions (`1.0.1.dev5058`), and it worked perfectly. Thanks a lot!! =)

Just in case, is there an expected release date for version 1.0.1??
